EmotiBitVersionController.cpp

```diff
@@ -421,4 +421,50 @@ int EmotiBitVersionController::readEmotiBitVersionFromSi7013()
 	uint8_t emotibitVersion = 0;
 	emotibitVersion = (uint8_t)_tempHumiditySensor.readRegister8(EMOTIBIT_VERSION_ADDR_SI7013_OTP, true);
 	return emotibitVersion;
+}
+
+// returns the version from barcode string "SKU-VERSION-NUMBER"
+int EmotiBitVersionController::getVersionFromBarcode(const char* barcode)
+{
+	char* temp;
+	char bcode[12];
+	strcpy(bcode, barcode);
+	char* version;
+	temp = strtok(bcode, "-"); // returns the sku
```

Comment:
we should change this. Right now, it depends on the barcode format. We should atleast make a another variable, that can track the barcode format, and change the parsing based on the format of the barcode